src/test/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivityTest.java

```diff
@@ -164,4 +171,168 @@ public void getLocationPermissions_withPermissionsAlreadyDenied_returnsFalse() {
 				});
 		}
 	}
+
+	@Test
+	public void onTouchEvent_withVolumeBtnPressed_startsSettingsDialogActivity() {
+		scenarioRule
+			.getScenario()
+			.onActivity(embeddedBrowserActivity -> {
+				Intents.init();
+				//> GIVEN
+				MotionEvent event = mock(MotionEvent.class);
+				when(event.getPointerCount()).thenReturn(2);
+				when(event.getActionMasked()).thenReturn(MotionEvent.ACTION_POINTER_DOWN);
+				when(event.getX(0)).thenReturn((float) 262.99);
+				when(event.getX(1)).thenReturn((float) 280.49);
+
+				embeddedBrowserActivity.onKeyDown(KeyEvent.KEYCODE_VOLUME_DOWN, null);
+				OnTouchListener onTouchListener = embeddedBrowserActivity.onTouchEvent();
+
+				//>THEN
+				onTouchListener.onTouch(null, event);
+
+				when(event.getActionMasked()).thenReturn(MotionEvent.ACTION_MOVE);
+				when(event.getX(0)).thenReturn((float) 562.99);
+				when(event.getX(1)).thenReturn((float) 580.49);
+
+				onTouchListener.onTouch(null, event);
+
+				//> WHEN
+				Intent intent = Intents.getIntents().get(0);
+				assertNotNull(intent);
+				assertEquals(
+					SettingsDialogActivity.class.getName(),
+					intent.getComponent().getClassName()
+				);
+				Intents.release();
+			});
+	}
+
+	@Test
+	public void onTouchEvent_withVolumeBtnReleasedBeforeSwipe_doesntStartActivity() {
+		scenarioRule
+			.getScenario()
+			.onActivity(embeddedBrowserActivity -> {
+				Intents.init();
+				//> GIVEN
+				MotionEvent event = mock(MotionEvent.class);
+				when(event.getPointerCount()).thenReturn(2);
+				when(event.getActionMasked()).thenReturn(MotionEvent.ACTION_POINTER_DOWN);
+				when(event.getX(0)).thenReturn((float) 262.99);
+				when(event.getX(1)).thenReturn((float) 280.49);
+
+				embeddedBrowserActivity.onKeyDown(KeyEvent.KEYCODE_VOLUME_DOWN, null);
+				OnTouchListener onTouchListener = embeddedBrowserActivity.onTouchEvent();
+				embeddedBrowserActivity.onKeyUp(KeyEvent.KEYCODE_VOLUME_DOWN, null);
```

Comment:
Minor, but it might be nice to move this line down to come right before the second call to `onTouch` (to additionally demonstrate that the volume button has to be held down through the entire duration of the swipe gesture).